### PR TITLE
Remove deprecated unused option in seed job script

### LINKED ISSUE #22184: [Bug]: Seed Job failing with AttributeError for 'ldap' module
### What happened?

e.g. https://ci-beam.apache.org/job/beam_SeedJob/9949/console

```
INFO:root:Generating Committers.groovy
Traceback (most recent call last):
  File "main.py", line 143, in <module>
    generate_groovy(args.output_dir, args.ldap_uris, args.cert_path)
  File "main.py", line 53, in generate_groovy
    github_usernames=get_committers_github_usernames(
  File "main.py", line 66, in get_committers_github_usernames
    ldap.set_option(ldap.OPT_X_TLS, ldap.OPT_X_TLS_DEMAND)
AttributeError: module 'ldap' has no attribute 'OPT_X_TLS'
+ echo 'ERROR: Failed to fetch committers'
ERROR: Failed to fetch committers
```

Also affects standalone seed job (https://ci-beam.apache.org/job/beam_SeedJob_Standalone/8489/console)

### Issue Priority

Priority: 2

### Issue Component

Component: testing

### PR DESCRIPTION
**Pull request by Abacn**
Fixes #22184

**Please** add a meaningful description for your change here

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by Abacn** [comment]
R: @ahmedabu98

**Comment by Abacn** [comment]
R: @damccorm 

Needs Committer to run seed job. Thanks!

File: .test-infra/jenkins/committers_list_generator/main.py [hunk 1]
```diff
@@ -63,7 +63,6 @@ def get_committers_github_usernames(ldap_uris, cert_path):
     connection = None
     try:
         ldap.set_option(ldap.OPT_X_TLS_CACERTFILE, cert_path)
-        ldap.set_option(ldap.OPT_X_TLS, ldap.OPT_X_TLS_DEMAND)
```
[the remark this answers is not in the document]
Comment (Abacn) [reply .test-infra/jenkins/committers_list_generator/main.py:66]: According to this https://github.com/python-ldap/python-ldap/issues/67 "OPT_X_TLS is deprecated and no longer documented in OpenLDAP: https://www.openldap.org/lists/openldap-software/200409/msg00617.html" (which says "This ldap_set_option(LDAP_OPT_X_TLS) feature is used to explicitly establish an ldaps session. The use of ldaps comes from LDAP version 2 and is deprecated;")

[the remark this answers is not in the document]
Comment (damccorm) [reply .test-infra/jenkins/committers_list_generator/main.py:66]: Ah, I see - so now it should be inferred and we don't need the explicit option (and it throws). Thanks for the context, just wanted to make sure we weren't losing something we'd need to replace

**Pull request merged**